Fix SSE tests and emit new "beforeMessage" event for SSE Extension

neelrr1 opened the pull request:
## Description
I'm building a toy chat app with HTMX + the SSE extension. I have some custom JS that I want to run **before and after** new SSE messages are swapped in from the server.

For a regular request with HTMX, I could add listeners to the `htmx:beforeSwap` and `htmx:afterSwap` events. Similarly, the WS extension emits `htmx:wsBeforeMessage` and `htmx:wsAfterMessage`, but there is no corresponding ability to listen to "before swap" messages with the SSE extension.

This PR introduces a new event, `htmx:sseBeforeMessage`, that is emitted before the SSE extension swaps in content from the server. Open to feedback on naming convention!

## Testing
I tested this change manually by applying this patch to my local copy of the SSE extension.

## Checklist

* [x] I have read the contribution guidelines
* [x] I have targeted this PR against the correct branch (`master` for website changes, `dev` for
  source changes)
* [x] This is either a bugfix, a documentation update, or a new feature that has been explicitly
  approved via an issue
* [x] I ran the test suite locally (`npm run test`) and verified that it succeeded

neelrr1 commented:
It may also make sense to rename the `sseMessage` event to `sseAfterMessage` but that would be a breaking change.

fhp-mec commented:
Hi,

Given that the naming is at least consistent with the ws version it seems like the best choice for the name of the event if thats what is decided to be the best solution.

That said, I do think that the future of the sse extension would probably be to use htmx's built in swap function rather than the bespoke one that is currently being used, and therefore emit both before and after swap events consistent with the existing htmx behavior. (see https://github.com/bigskysoftware/htmx/issues/844 for context)

If having `beforeSwap` and `afterSwap` events be fired by the swapping function would give you the functionality you need I would suggest that instead of adding a new event. 

I would also suggest that you add a regression test to the testing suite no matter what the solution is.

Renerick commented on src/ext/sse.js:
```diff
@@ -148,6 +148,7 @@ This extension adds support for Server Sent Events to htmx.  See /www/extensions
 					}
 
 					// swap the response into the DOM and trigger a notification
+					api.triggerEvent(elt, "htmx:sseBeforeMessage", event);
```
This probably should provide cancellation support by wrapping this call in an if statement, like so 
https://github.com/bigskysoftware/htmx/blob/master/src/ext/ws.js#L131-L136

Also, would be nice to add tests for this event (and its cancellation)

neelrr1 commented:
@Renerick, thanks so much for the first review and your comments. I've updated the code to add cancellation support and added tests to verify that the events are being emitted correctly. Can you take another look?

In the process, I also fixed our test suites for the SSE extension and `hx-sse` attribute. These were not running due to syntax errors in the test files, (#2167) which hid the fact that the tests were broken.

Renerick reviewed (approved):
LGTM

Especially thank you for fixing syntax errors in SSE tests

Renerick reviewed (approved):
LGTM

Special thank you for fixing syntax errors in SSE tests 🙂

neelrr1 commented:
@Renerick are you able to run CI so we can get this merged?
